src/learning/gp.rs

```diff
@@ -193,11 +203,11 @@ impl<T: Kernel, U: MeanFunc> GaussianProcess<T, U> {
 
             let v_mat = Matrix::new(test_mat.rows(), test_mat.cols(), var_data);
 
-            let post_var = self.ker_mat(inputs, inputs) - &v_mat * v_mat.transpose();
+            let post_var = try!(self.ker_mat(inputs, inputs)) - &v_mat * v_mat.transpose();
 
-            return (post_mean, post_var);
+            Ok((post_mean, post_var))
+        } else {
+            Err(Error::new(ErrorKind::UntrainedModel, "The model has not been trained."))
```

Comment:
I more or less copied the std::io implementation but I agree that this case doesn't really fit in. I'll see how easy it is to modify - but it seems this particular error will be disappearing if we modify the model traits.